### ISSUE TITLE
SMILES validation

### ISSUE THREAD

**Issue by cmgrulke** [Chemical-Curation/chemcurator_django#30]
**As a** researcher
**I want** the API to check my SMILES format is correct
**So that** if the format I try to submit as SMILES is not SMILES, it provides me an error

**Request Example**
`POST https://api.chemreg.epa.gov/defined-compounds/`

**Acceptance Criteria:**
When I post a SMILES to the API, it verifies that the provided string is a SMILES
If the string not SMILES, the input is invalid (415 or 400?) (Message: "Smiles format is invalid.").

**Test:**

**Comment by cmgrulke** [comment]
I recommend that a SMILES string is checked with a regular expression to attempt to identify things that are not meeting SMILES format.  There is no one defined SMILES format right now, it is essentially a series of closely related implementations all similar, but not always clearly documented.  Documentation of one flavor of SMILES can be accessed here (http://opensmiles.org/opensmiles.pdf)
I think @asif-epa has a regexp he found to perform this check

**Comment by cmgrulke** [comment]
Some example SMILES:
CC(=O)OC1=C(C=CC=C1)C(O)=O
CC(=O)NC1=CC=C(O)C=C1
CC(C)(C1=CC=C(O)C=C1)C1=CC=C(O)C=C1
ClC(Cl)=C(Cl)Cl
ClC=C(Cl)Cl
C=O
[S]
CCNC1=NC(NC(C)C)=NC(Cl)=N1
CC(=O)N1CCN(CC1)C1=CC=C(OC[C@H]2CO[C@@](CN3C=CN=C3)(O2)C2=CC=C(Cl)C=C2Cl)C=C1

**Comment by mwfrost** [comment]
Chris and I discussed this validation process in the Slack while I worked on issue #29  , and he agreed that using the `partialsmiles` library was a better option than using a regex. The branch for #29 includes a validator `validate_smiles` that checks whether `ParseSmiles(smiles, partial=False)` can turn the string `smiles` into a structure. 

By default the `PartialSmiles` class rejects uncommon valences or charge states like `[S]`, but Chris says such expressions are normal for our data. Therefore `validate_smiles` explicitly passes on any `ValenceError` instances and only raises `SMILESSyntaxError` instances.
